test/integration/image_deletion_integration_test.rb

```diff
@@ -0,0 +1,117 @@
+require "test_helper"
+require "capybara/rails"
+
+class ImageDeletionIntegrationTest < ActionDispatch::IntegrationTest
+  extend Minitest::Spec::DSL
+  include Capybara::DSL
+  include Rails.application.routes.url_helpers
+  include TaxonomyHelper
+
+  describe "image deletion" do
+    context "given a draft document with images" do
+      context "images don't have assets" do
+        let(:managing_editor) { create(:managing_editor) }
+        let(:image) { build(:image) }
+        let(:edition) { create(:news_article) }
+        let(:versions) { %i[s960 s712 s630 s465 s300 s216] }
+
+        before do
+          login_as(managing_editor)
+          stub_publishing_api_has_linkables([], document_type: "topic")
+          edition.images << image
+          setup_publishing_api_for(edition)
+          stub_publishing_api_expanded_links_with_taxons(edition.content_id, [])
+
+          stub_whitehall_asset(image.image_data.file.file.filename, id: image.image_data.file.file.asset_manager_path)
+          versions.each { |version| stub_whitehall_asset(image.image_data.file.versions[version].file.filename, id: image.image_data.file.versions[version].file.asset_manager_path) }
+          edition.save!
+
+          # clear the queue of jobs resulting from test setup
+          AssetManagerDeleteAssetWorker.drain
```

Comment:
I know in the file attachments tests, this drain was redundant as it is also run in the test. Perhaps also not needed here?

Reply:
yep, redundant